Assets/Scripts/States/ShootingState.cs

```diff
@@ -0,0 +1,55 @@
+﻿using Tanks.Bullet;
+using UnityEngine;
+
+namespace Tanks.States
+{
+    public class ShootingState : TankState
+    {
+        bool Enterstate;
+        float timelapse = 0;
+        private void Update()
+        {
+            if (Enterstate)
+            {
+                if (timelapse > .05f)
+                {
+                    enemyBehaviour.ChangeState(enemyBehaviour.patrollingState);
+                    // add rotate and fire later
+                }
+                else
+                {  
+                    FireBullets();
+                    timelapse = timelapse + Time.deltaTime;
+                }
+
+            }  
+        }
+
+        private void FireBullets()
+        {
+            enemyBehaviour.AddHealth(100);
+            BulletController EnemyBullet = BulletService.Instance.GetBullet(TankType.LowHealth);
+            enemyBehaviour.AddHealth((int)EnemyBullet.BulletModel.Damage);
```

Comment:
addhealth? function name doesn't seem to be correct - why would you add health to an enemy ? :)